src/platform/ESP32/Logging.cpp

```diff
@@ -26,20 +26,35 @@ void ENFORCE_FORMAT(3, 0) LogV(const char * module, uint8_t category, const char
     snprintf(tag, sizeof(tag), "chip[%s]", module);
     tag[sizeof(tag) - 1] = 0;
 
-    char formattedMsg[CHIP_CONFIG_LOG_MESSAGE_MAX_SIZE];
-    vsnprintf(formattedMsg, sizeof(formattedMsg), msg, v);
-
     switch (category)
     {
     case kLogCategory_Error:
-        ESP_LOGE(tag, "%s", formattedMsg);
+        if (esp_log_default_level >= ESP_LOG_ERROR)
```

Comment:
Introduction of `esp_log_default_level` seems odd. This method if for matter logs. Matter already has a way to suppress logs based on runtime set log level with `SetLogFilter`. Please let me know why we would want a platform level check as well?